src/main/java/org/apache/commons/dbcp2/BasicDataSource.java

```diff
@@ -1939,6 +1940,9 @@ public synchronized void close() throws SQLException {
             final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
             try {
                 mbs.unregisterMBean(registeredJmxName);
+            } catch (final InstanceNotFoundException infe) {
+                // Suppress warning and exception. It is often the case that MBeans are
+                // de-registered before the close method is invoked.
```

Comment:
Hi again and thank you for your changes. I forgot to ask why not code this more 'cleanly' by first calling MBeanServer.isRegistered() instead of catching an exception.